src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/ITestPlatform.cs

```diff
@@ -68,7 +68,9 @@ public interface ITestPlatform : IDisposable
         /// </param>
         /// <param name="criteria">Specifies the start test session criteria.</param>
         /// <param name="eventsHandler">Events handler for handling session events.</param>
-        void StartTestSession(
+        ///
+        /// <returns>True if the operation succeeded, false otherwise.</returns>
+        bool StartTestSession(
```

Comment:
should this also be marked as obsolete for now?

also, will this block if the testhost could not be spawned right away? because of CPU constraints?

Reply:
Not really because the entry points are already marked as obsolete (i.e. the APIs in the Translation Layer). Even though the API pointed here is public, I'm not aware of anyone actually implementing their own TestPlatform by means of using the `ITestPlatform` interface. Even if they did I'm not sure how that would work (i.e., how would we even discover a custom implementation of TP). As far as I remember this is not even marked as an extension point.